core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/part/EnvelopePart.kt

```diff
@@ -686,9 +687,14 @@ fun minEnvelopeParts(
                         pos,
                         speedB
                     )
-                // Add intersection point
-                newPositions.add(intersection.position)
-                newSpeeds.add(intersection.speed)
+                // Add intersection point if it isn't either the previous or the next position.
```

Comment:
From the inital test `(prevSpeedA - prevSpeedB) * (speedA - speedB) < 0`, `prevSpeedA`&`B` can't be equal and same for `speedA`&`B`, no?
My initial idea was to short-circuit intersection search early, but it looks like it is already OK... unless there are some float precision in the equation and we could use an epsilon ?

also nit:
```suggestion
                // Add intersection point if it isn't either the previous or the current position.
```

Reply:
Technically true, but basically it's because of an epsilon yeah. This created a bug where the envelope wouldn't be strictly monotonous.
Basically, the interpolation would compute and add an intersection position at 13000.000000000001, but the actual next position was 13000. So I just don't add the intersection if the position is less than an epsilon away from the intersection.

Reply:
Then can't you just change initial test to `(prevSpeedA - prevSpeedB) * (speedA - speedB) < -SPEED_EPSILON` (or even square it) and remove that change?
It's only a little win for code and for perf, feel free to resolve.

Reply:
I feel like there could still be hiccups if i do this (and yeah it would definitely be speed_epsilon squared). I'd rather just double check the position.